### PR TITLE
Update Waypoint documentation overview page

### PR DESCRIPTION
**Pull request by xiaolin-ninja**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-xx-update-waypoint-fundamentals-hashicorp.vercel.app/waypoint/docs) 🔎

## 🗒️ What

Update Waypoint documentation landing page to highlight runners.

## 🤷 Why

Customers are confused about runners, and you can't use Waypoint without them.

### CODE REVIEW

**Review by kendallstrautman — APPROVED** [review]

**Comment by xiaolin-ninja** [comment]
Can I merge this PR, or should I wait for someone on the team to merge it?

**Comment by kendallstrautman** [comment]
@xiaolin-ninja sorry just saw your comment on the PR! Once you have an approval from a dev-portal team member, feel free to merge it. I'll update your branch and set it for auto-merge now.

**Pull request merged**
